### PR TITLE
Refactor Tests

### PR DESCRIPTION
**Pull request by SepidehAlassi**
Resolves [DSP-466](https://dasch.myjetbrains.com/youtrack/agiles/110-4/111-36?issue=DSP-466)

Todo:
- [x] move some checks from E2E tests to normal Specs
- [x] add missing BUILD.bazel files
- [x] correct rendering of fenced code blocks in admin API docs

### CODE REVIEW

File: webapi/src/test/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = [
+        "//webapi:main_library",
+        "//webapi:test_library",
+        "@maven//:org_scalatest_scalatest_2_12",
+        "@maven//:org_scalactic_scalactic_2_12",
+    ],
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/BUILD.bazel:23]: can you please always add a new line at the end of a file?

File: webapi/src/test/scala/org/knora/webapi/messages/admin/responder/groupsmessages/GroupsMessagesADMSpec.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                  description = Some("A new group with an invalid custom Iri"),
+                  project = SharedTestDataADM.IMAGES_PROJECT_IRI,
+                  status = true,
+                  selfjoin = false
+                )
+            )
+            assert(caught.getMessage === "Invalid group IRI")
+        }
+    }
+}
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/admin/responder/groupsmessages/GroupsMessagesADMSpec.scala:55]: newline missing

File: webapi/src/test/scala/org/knora/webapi/messages/admin/responder/projectsmessages/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        "//knora-ontologies",
+        "//test_data",
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = [
+        "//webapi:main_library",
+        "//webapi:test_library",
+    ] + BASE_TEST_DEPENDENCIES_WITH_JSON,
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/admin/responder/projectsmessages/BUILD.bazel:22]: newline missing

File: webapi/src/test/scala/org/knora/webapi/messages/admin/responder/usersmessages/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        "//test_data",
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = [
+        "//webapi:main_library",
+        "//webapi:test_library",
+        "@maven//:org_springframework_security_spring_security_core",
+    ] + BASE_TEST_DEPENDENCIES_WITH_JSON,
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/admin/responder/usersmessages/BUILD.bazel:23]: newline missing

File: webapi/src/test/scala/org/knora/webapi/messages/store/triplestoremessages/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    data = [
+        "//knora-ontologies",
+        "//test_data",
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = [
+        "//webapi:main_library",
+    ] + BASE_TEST_DEPENDENCIES_WITH_JSON,
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/store/triplestoremessages/BUILD.bazel:21]: newline missing

File: webapi/src/test/scala/org/knora/webapi/messages/v1/responder/permissionmessages/BUILD.bazel [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = [
+        "//webapi:main_library",
+        "//webapi:test_library",
+        "@maven//:org_scalatest_scalatest_2_12",
+        "@maven//:org_scalactic_scalactic_2_12",
+    ],
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/v1/responder/permissionmessages/BUILD.bazel:23]: newline missing

File: webapi/src/test/scala/org/knora/webapi/messages/v1/responder/usermessages/BUILD.bazel [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = [
+        "//webapi:main_library",
+        "//webapi:test_library",
+        "@maven//:org_scalatest_scalatest_2_12",
+        "@maven//:org_scalactic_scalactic_2_12",
+        "@maven//:org_springframework_security_spring_security_core",
+    ],
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/v1/responder/usermessages/BUILD.bazel:24]: newline missing

File: webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/BUILD.bazel [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        "//knora-ontologies",
+        "//test_data",
+    ],
+    jvm_flags = ["-Dconfig.resource=fuseki.conf"],
+    # unused_dependency_checker_mode = "warn",
+    deps = [
+        "//webapi:main_library",
+        "//webapi:test_library",
+    ] + BASE_TEST_DEPENDENCIES,
+)
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/v2/responder/ontologymessages/BUILD.bazel:22]: newline missing

**Comment by SepidehAlassi** [comment]
@subotic Thanks for reviewing this. Newlines are now added to the end of the files.

**Pull request merged**